### DISCUSSION TITLE
how to use index signature as type

### DISCUSSION THREAD

**Discussion by schavan-rsi** [MichalLytek/type-graphql#1355] [Q&A] [answered]
`@ObjectType()
export class Result {
  @Field({ nullable: true })
  domainName?: string;

  @Field({ nullable: true })
  uniqueId?: string;

  @Field({ nullable: true })
  resultKeyVal: { [key: string]: string };
}`

getting below error: Unable to infer GraphQL type from TypeScript reflection system. You need to provide explicit type for 'resultKeyVal' of 'Result' class.

I found this below discussion but it won't solve my issue: 
https://github.com/MichalLytek/type-graphql/discussions/1303

**Comment by MichalLytek** [comment, marked as answer]
It will solve your issue and it's the only way to express it in GraphQL.

**Reply by schavan-rsi** [reply]
Thank you for your reply. Please consider object is like this one (I can add any key-value in that object):
```
const obj: {[key: string]: string} = {
    id: '1',
    fname: 'Sujit',
    lname: 'Chavan',
   // here we have too many dynamic key-values
   }
```

how to accept that in type-graphql?

**Reply by MichalLytek** [reply]
```ts
@ObjectType()
export class StringKeyStringValue {  
	@Field()
	key: string;
	
	 @Field()
    value: string;
}

@ObjectType()
export class Result {    
	@Field(() => [StringKeyStringValue])
	resultKeyVal: StringKeyStringValue[];
}
```

```ts
const result = {
  resultKeyVal: [{ key: 'id', value: '1' }, { key: 'fname', value: 'Sujit' }, { key: 'lname', value: 'Chavan' }],
}
```

**Reply by MichalLytek** [reply]
basically it's like `Object.entries(obj).map(([key, value]) => ({ key, value }))`

**Reply by MichalLytek** [reply]
or if you don't care about if it's array, string or number, just use any graphql json scalar

**Reply by schavan-rsi** [reply]
No luck with above solution.

if we don't know the type exactly, then what we can use as general type?
```
@ObjectType()
export class Result {    
	@Field()
	resultKeyVal: any;
}
```

**Reply by MichalLytek** [reply]
any graphql json scalar

**Reply by schavan-rsi** [reply]
GraphQLJSONObject works for me thanks!
